### PR TITLE
bollgrid: generate the last order if balance is not enough

### PR DESCRIPTION
**Pull request by LarryLuTW**
In bollgrid strategy, The original behavior is that bbgo will stop generating new orders when the balance is insufficient.

Although the left balance is not much, I think it's wasteful to be idle there. Therefore, I use the remaining balance to send the last order when the balance is insufficient. I hope this will increase the utilization of assets.

### CODE REVIEW

**Comment by zenixls2** [comment]
The total value of the order must be over 10usd in most platforms.  
Be sure to check the remaining balance is above that.

**Comment by c9s** [comment]
btw there is a bbgo.AdjustQuantityXXXX function you can use

File: pkg/strategy/bollgrid/strategy.go [hunk 1]
```diff
@@ -226,7 +228,9 @@ func (s *Strategy) generateGridSellOrders(session *bbgo.ExchangeSession) ([]type
 		}
 		baseQuantity := fixedpoint.NewFromFloat(order.Quantity)
 		if baseBalance < baseQuantity {
-			log.Infof("base balance %f is not enough, stop generating sell orders", baseBalance.Float64())
+			log.Infof("base balance %f is not enough, generating the last sell orders", baseBalance.Float64())
+			order.Quantity = baseBalance.Float64()
```
Comment (c9s) [inline pkg/strategy/bollgrid/strategy.go:232]: try:
```
quantity = bbgo.AdjustQuantityByMaxAmount(quantity, closePrice, b.Available.Float64())
```

**Comment by LarryLuTW** [comment]
@c9s Please help me review this commit. My current approach is to adjust the quantity before constructing the order. If the adjusted quantity is too small, I will not send the order.

**Review by c9s — APPROVED** [review]
looks great! thanks!

**Comment by c9s** [comment]
thank you for your contribution! 😎 if you're interested in BBG, please check the link below:

http://bit.ly/get-your-bbg

**Pull request merged**